common/src/main/java/com/wynntils/utils/mc/StyledTextUtils.java

```diff
@@ -198,4 +207,26 @@ public static StyledText unwrap(StyledText styledText) {
 
         return StyledText.fromParts(newParts);
     }
+
+    public static Pair<String, String> extractNameAndNick(StyledText styledText) {
```

Comment:
I'll mark this for myself tomorrow. I may or may not want this in a model, rather than this util, as it's too specific to StyledText. It's likely that a model of some sort should hold this code.

Reply:
Fair enough I'll leave that to you then. I believe I've done what was necessary for this util function to work as expected.

With regard to the error if a major id was moved, if the major id was moved then the first level of the map in the code would still have Hollowed as a key but it would have 2 as the secondary map's key rather than 3 so the second get would return null and everything would continue as expected. calling get() on a map where the key isn't present simply returns null rather than raising an error. I believe a containsKey() check is implicit to the get call

Reply:
Yeah, this could just live in RaidModel, right?

Reply:
Although I can't think of any currently, there's probably other places where getting username is useful so RaidModel could work for now but not a permenant place for it imo

Reply:
It's definitely a util method rather than something that is exclusive to this Model or part of the game. Granted those use cases haven't presented themselves yet so I'm happy to move it to the RaidModel until that point if that's the consensus

Reply:
Okay, let's keep this here for now.